cursorless-snippets/functionDeclaration.cursorless-snippets

```diff
@@ -8,6 +8,10 @@
             "typescriptreact",
             "javascript",
             "javascriptreact"
+          ],
+          "scopeTypes": [
+            "namedFunction",
+            "document"
```

Comment:
I had to do this to make sure that this snippet wins when you're inside a function that is inside a class.  Alternately, we could add some new field to the member function snippet definition below that causes it not to match when its nearest ancestor is a function, not a class.  Not exactly sure the mechanism there

The disadvantage of this current approach is that it makes it a bit harder for users to override, because if they don't match these `scopeTypes` exactly then the core snippet here could just win because it's more specific 🤔